### PR TITLE
Feature/uc3 modified spat data

### PR DESCRIPTION
**Pull request by abey-yoseph**
Adding plot for all signal group event states vs time.

# PR Details
## Motivation and Context
Required UC3 plots

## How Has This Been Tested?
Tested with simulation data.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-analytics-fotda/blob/main/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tsmo/src/modified_spat_plotter.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+   
+    all_in_filenames = os.listdir(input_directory_path)
+    for file in all_in_filenames:
+        #read the parsed spat data file generated by the modified_spat_parser script
+        if "modified_spat" in file and "parsed" in file:
+            modified_spat_data = pd.read_csv(f'{input_directory_path}/{file}')
+            signal_groups = modified_spat_data['Signal_Group'].unique()
+
+            #Use moy and timestamp fields to get epoch time for each record
+            first_day_epoch = dt.datetime(2022, 1, 1, 0, 0, 0).timestamp() * 1000 #get time since epoch for beggining of year in ms
```
Comment (paulbourelly999) [inline tsmo/src/modified_spat_plotter.py:28]: This script will break when the year changes. You may want to retrieve the current year.

**Review by Saeid-Solei — APPROVED** [review]

File: tsmo/src/status_intent_plotter.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def status_intent_plotter():
 
         plt.xticks(rotation=75)
         axs=plt.gca()
-        xfmt = md.DateFormatter('%H:%M:%S.%f') 
+        xfmt = md.DateFormatter('%H:%M:%S')
```
Comment (paulbourelly999) [inline tsmo/src/status_intent_plotter.py:63]: Did you remove the decimal accuracy? This is fine as long as you don't have sequential ticks that look like the represent the same time.

File: tsmo/src/modified_spat_plotter.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                        time2 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i+1])
+                        ax1.hlines(group, time1, time2, color='yellow', linewidth=10)
+                    #change in state from red to green, draw green
+                    elif (group_subset['Event_State'].iloc[i] == 3)&(group_subset['Event_State'].iloc[i+1] == 6):
+                        time1 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i])
+                        time2 = dt.datetime.fromtimestamp(group_subset['Epoch_Time(s)'].iloc[i+1])
+                        ax1.hlines(group, time1, time2, color='green', linewidth=10)
+
+                        #add vertical line for green state change
+                        ax1.vlines(time2, 0, 9, linestyles='dashed', color='green', linewidth=1)
```
Comment (paulbourelly999) [inline tsmo/src/modified_spat_plotter.py:67]: I thought we removed these green vertical lines. Where are you place these. Just at green starts/end. If so you can probably remove this vertical line.

Comment (abey-yoseph) [reply tsmo/src/modified_spat_plotter.py:67]: I left them for now, but yea I think the plan for the future is to have the ET zones displayed for each vehicle.

File: tsmo/src/status_intent_plotter.py (hunk 2 above)
Comment (abey-yoseph) [reply tsmo/src/status_intent_plotter.py:63]: Removed them per @Saeid-Solei request. Yes there aren't any of those kind of ticks.

**Pull request merged**